08_education/preschool_county.qmd

```diff
@@ -0,0 +1,766 @@
+---
+title: "Access to pre-kindergarten Predictor, County-level"
+author: "JP Walsh"
+date: today
+format:
+  html:
+    toc: true
+    toc_float: true
+    embed-resources: true
+    code-fold: show
+execute: 
+  warning: false
+editor_options: 
+  chunk_output_type: console
+---
+
+*Program for the creation of the Access to Pre-Kindergarten Upward Mobility Predictor at the County-level*
+
+ACS Code: Initially written by Tina Chelidze and updated by JP Walsh 2022-2023.
+
+Primary data derived from the IPUMS API.
+
+Based on processes developed by Paul Johnson and Kevin Werner in SAS.
+
+## Housekeeping
+
+Read in packages for use. If you do not have an IPUMS API key you will need to make one via the [IPUMS website](https://www.ipums.org/).
+
+```{r, results='hide'}
+library(tidyverse)
+library(Hmisc)
+library(ipumsr)
+library(reactable)
+library(srvyr)
+library(scales)
+
+options(scipen = 999)
+
+
+# DO NOT PUSH YOUR API KEY. You only have to run this once and then comment it out as below.
+#set_ipums_api_key("Your KEY", save = TRUE)
+
+source(here::here("functions", "API", "extract_ipums.R"))
+source(here::here("functions", "API", "ipums_repwt_pre-k.R"))
+source(here::here("08_education", "R", "finalize_metric.R"))
+source(here::here("08_education", "R", "calc_survey_mean.R"))
+source(here::here("08_education", "R", "calc_survey_mean_subgroup.R"))
+```
+
+## Read ACS Data
+
+### Person-Level Data
+
+Read in the ACS extracts using the extract_ipums function. Make sure to change the survey list to reflect what years you want to include in the data. Currently we are including two years of 5-year data (2018c and 2021c). 5-year 2022 data will be added pending its release.
+```{r}
+acs <- extract_ipums(
+  extract_name = "umf_data_18_22",
+  extract_description = "Microdata pull for Access to Pre-K Metric Predictors. American Community Survey, years 2018, 2019 and 2022 (1 and 5-year).",
+  survey = list("us2018a", "us2021a")
+) 
```

Comment:
I just realized that this function saves the data in the data/ folder. This has traditionally only contained output data. We should probably clean this up (later).

Reply:
The extract description for this is incorrect.

Reply:
Actually, I think the extract is incorrect. It pulls the 1-year data.